Source/WebCore/Modules/compression/DecompressionStreamDecoder.cpp

```diff
@@ -147,4 +143,72 @@ ExceptionOr<Vector<uint8_t>> DecompressionStreamDecoder::decompress(const uint8_
 
     return output;
 }
+
+#ifdef __APPLE__
+ExceptionOr<bool> DecompressionStreamDecoder::initializeAppleCompressionFramework() 
+{
+    initailized = true;
+    
+    auto result = compression_stream_init(&stream, COMPRESSION_STREAM_DECODE, COMPRESSION_ZLIB);
+    if (result != COMPRESSION_STATUS_OK)
+        return Exception { TypeError, "Initialization Failed."_s };
+
+    return true;
+}
+
+ExceptionOr<Vector<uint8_t>> DecompressionStreamDecoder::decompressAppleCompressionFramework(const uint8_t* input, const size_t inputLength)
+{
+    size_t index = 0, allocateSize = startingAllocationSize, totalSize = 0;
+    Vector<Vector<uint8_t>>storage;
+    storage.reserveCapacity(128);
```

Comment:
I am not sure reserveCapacity is needed here.

Reply:
I was trying to reserve some space up front, so we don't need to keep reallocating the Vector of pointers.

I was using the reserveCapacity, so it didn't modify the size of vector just the capacity.

Reply:
Rewrote logic. Above comment does not apply anymore 😀